aggregator/src/aggregator/http_handlers/tests/report.rs

```diff
@@ -44,26 +45,35 @@ use trillium_tokio::Stopper;
 async fn upload_handler() {
     async fn check_response(
         test_conn: &mut TestConn,
-        desired_status: Status,
-        desired_type: &str,
-        desired_title: &str,
-        desired_task_id: &TaskId,
-        desired_detail: Option<&str>,
+        desired_report_id: &ReportId,
+        desired_report_error: ReportError,
     ) {
-        let mut desired_response = json!({
-            "status": desired_status as u16,
-            "type": format!("urn:ietf:params:ppm:dap:error:{desired_type}"),
-            "title": desired_title,
-            "taskid": format!("{desired_task_id}"),
-        });
-        if let Some(detail) = desired_detail {
-            desired_response
-                .as_object_mut()
-                .unwrap()
-                .insert("detail".to_string(), json!(detail));
+        // HTTP status is OK regardless of what happened to the constituent reports because the HTTP
+        // messages were exchanged successfully.
+        if test_conn.status() != Some(Status::Ok) {
+            println!(
+                "ERROR: Report {} got status {:?}",
+                desired_report_id,
+                test_conn.status()
+            );
+            if let Some(body) = test_conn.take_response_body_string() {
+                println!("Response body: {}", body);
+            }
+        }
+        assert_eq!(test_conn.status(), Some(Status::Ok));
+
+        assert_headers!(&test_conn, "content-type" => "application/dap-upload-resp");
+        let body = &take_response_body(test_conn).await;
+        let expected_content_len = format!("{}", body.len());
+        let len_str = expected_content_len.as_str();
+        assert_headers!(&test_conn, "content-length" => len_str);
+        let upload_response = UploadResponse::get_decoded_with_param(&body.len(), body).unwrap();
+
+        for status in upload_response.status() {
+            assert_eq!(status.report_id(), *desired_report_id);
+            assert_eq!(status.error(), desired_report_error);
+            assert_eq!(status.encoded_len(), Some(17)); // TKTK
         }
```

Comment:
Here's what I was thinking
```suggestion
        assert_eq!(upload_response.status().len(), 1);
        for status in upload_response.status() {
            assert_eq!(status.report_id(), *desired_report_id);
            assert_eq!(status.error(), desired_report_error);
        }
```